chronicle-queue/src/main/scala/akka/stream/alpakka/chroniclequeue/impl/ChronicleQueueBase.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+// ORIGINAL LICENCE
+/*
+ *  Copyright 2017 PayPal
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package akka.stream.alpakka.chroniclequeue.impl
+
+import java.io.File
+
+import akka.actor.{ActorSystem, Props}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler, StageLogging}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import com.typesafe.config.Config
+
+abstract class ChronicleQueueBase[T, S](
+    private[stream] val queue: PersistentQueue[T],
+    onPushCallback: () => Unit = () => {}
+)(implicit serializer: ChronicleQueueSerializer[T], system: ActorSystem)
+    extends GraphStage[FlowShape[T, S]] {
+
+  def this(config: Config)(implicit serializer: ChronicleQueueSerializer[T], system: ActorSystem) =
+    this(new PersistentQueue[T](config))
+
+  def this(persistDir: File)(implicit serializer: ChronicleQueueSerializer[T], system: ActorSystem) =
+    this(new PersistentQueue[T](persistDir))
+
+  private[stream] val in = Inlet[T]("PersistentBuffer.in")
+  private[stream] val out = Outlet[S]("PersistentBuffer.out")
```

Comment:
Since you renamed `PersistentBuffer` to `ChronicleQueue`, I suggest also renaming the `in` and `out` names.

Reply:
of course, thanks